File: Ex08/Ex08.py
k = int(input())

valores = []

valores = [int(x) for x in input().split()[:k]]

real = float(input())

print(f"SOMA_BRUTA: {sum(valores)}")
print(f"SOMA_ESCALADA: {sum(valores) * real:.2f}")

xorCheck = 0

for valor in valores:
    xorCheck^=valor

print(f"XOR_CHECK: {xorCheck}")

media = sum(valores) // k
print(f"MEDIA_INT: {media}")

qtdAcimaMedia = 0
primeiro = True

print("ACIMA_MEDIA:", end=" ")

for valor in valores:
    if valor > media:
        if not primeiro:
            print(" ", end="")
        print(valor, end="")
        primeiro = False
        qtdAcimaMedia += 1

if (qtdAcimaMedia == 0):
    print("NENHUM", end="")

print()

if xorCheck == 0:
    print("STATUS: OK")
else:
    print(f"STATUS: ERRO {xorCheck}")

File: Ex08/Ex08.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    
    int k;

    scanf("%d", &k);

    int* valores = (int*) malloc(sizeof(int) * k);
    
    for (int *p = valores; p < valores + k; p++) {
        scanf("%d", p);
    }

    float real;
    
    scanf("%f", &real);

    int soma = 0;

    for (int* p = valores; p < valores + k; p++) {
        soma += *p;
    }

    printf("SOMA_BRUTA: %d\n", soma);
    printf("SOMA_ESCALADA: %.2f\n", soma * real);

    int xor_check = 0;

    for (int* p = valores; p < valores + k; p++) {
        xor_check ^= *p;
    }

    printf("XOR_CHECK: %d\n", xor_check);
    int media = soma / k;
    printf("MEDIA_INT: %d\n", media);

    int qtd_acima_media = 0;

    int primeiro = 1;
    printf("ACIMA_MEDIA: ");

    for (int* p = valores; p < valores + k; p++) {
        if (*p > media) {
            if (!primeiro) {
                printf(" ");
            }
            printf("%d", *p);
            primeiro = 0;
            qtd_acima_media++;
        }
    }

    if (qtd_acima_media == 0) {
        printf("NENHUM");
    }

    printf("\n");

    if (xor_check == 0) {
        printf("STATUS: OK\n");
    } else {
        printf("STATUS: ERRO %d\n", xor_check);
    }

    free(valores);

    return 0;
}